transcom/mymove
# INT B-21022 - Include/Exclude specific offices

## [B-21022](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1006502&RoomContext=TeamRoom%3A855664&concept=TeamRoom)

## Summary

**INTEGRATION**

This PR addresses issue of certain offices showing and not showing in the close out widget for both customer and office apps when onboarding/editing PPMs.  An analysis was done to compare records in the transportation_offices table vs Excel/csv File:
[Transportation Offices with Statuses 5 Sep 2024  -Validations.csv](https://github.com/user-attachments/files/17032507/Transportation.Offices.with.Statuses.5.Sep.2024.-Validations.csv) with respect to the provides_ppm_closeout field/column. Per Danny M, the Excel/csv file serves as the Gold standard. 

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test
A migration file was created to update 7 records in the transporation_offices table, updating the provides_ppm_closeout column to match the Excel/csv file. They are the following:

PPPO Fort Belvoir - USA => change FALSE to TRUE
Personal Property Activity HQ (PPA HQ) - USAF => change TRUE to FALSE
PPPO Base Miami - USCG => change FALSE to TRUE
JPPSO - North West (JEAT) - USA  =>  change TRUE to FALSE
PPPO NSWC Panama City Division - USN  =>  change FALSE to TRUE
JPPSO - Mid Atlantic (BGAC) - USA => change TRUE to FALSE
JPPSO - South West (LKNQ) - USN => change TRUE to FALSE

TRUE => Visible in Closeout widget search results (selectable)
FALSE => Not shown/hidden in Closeout widget search results (not selectable)


1. Login as customer and create and submit a move containing a PPM.
2. Specifically verify the 7 offices are visible or hidden based on boolean state(provides_ppm_closeout) for the closeout widget. If you want you can do spot checks using anyone one of the offices in the attached Excel file according to the **_PPM Closeout_** column.
3. Login as service counselor and view submitted move PPM. Again verify the 7 offices are either visible or hidden on edit. 
4. Add a new PPM from SC. Do the same verification and spot checks using Excel file as guide.
5. Are there any other places missed and not specified in the instructions? If so, check that too.

![Screenshot 2024-09-17 at 11 41 05 AM](https://github.com/user-attachments/assets/5207e8bd-4d2f-454f-b21f-67b1a35d88cb)


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:
